File: app/src/main/java/com/dummyapp/viewpager/ViewPagerAdapter.java
package com.dummyapp.viewpager;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.List;

/**
 * Created by cl-macmini-86 on 8/18/17.
 */

public class ViewPagerAdapter extends FragmentPagerAdapter {
    private List<ViewPagerFragment> fragmentList;
    private List<String> titleList;


    public ViewPagerAdapter(final FragmentManager fm) {

        super(fm);

    }

    @Override
    public Fragment getItem(final int position) {
        ViewPagerFragment fragment = fragmentList.get(position);
        fragment.setPosition(position);
        return fragmentList.get(position);
    }

    @Override
    public int getCount() {
        return fragmentList.size();
    }

    public void setList(List<ViewPagerFragment> fragmentList, final List<String> titleList) {
        this.fragmentList = fragmentList;
        this.titleList = titleList;
    }

    @Override
    public CharSequence getPageTitle(final int position) {
        return titleList.get(position);
    }
}

File: app/src/main/java/com/dummyapp/kotlin/viewpager/ViewPagerAdapter.kt
package com.dummyapp.kotlin.viewpager

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

/**
 * Created by cl-macmini-86 on 9/26/17.
 */
class ViewPagerAdapter(fragmentManager: FragmentManager) : FragmentPagerAdapter(fragmentManager) {

    private var fragmentList: List<ViewPagerFragment>? = null


    override fun getItem(position: Int): Fragment {

        val fragment = fragmentList?.get(position)
        fragment?.setVluePosition(position)
        return fragmentList?.get(position)!!
    }

    override fun getCount(): Int {

        return fragmentList?.size ?: 0
    }

    fun setList(fragmentsList: MutableList<ViewPagerFragment>) {
        this.fragmentList = fragmentsList
    }
}